lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -18,7 +18,12 @@ module HaveHttpStatus
         # @return response matcher instance
         def self.matcher_for_status(target)
           if GenericStatus.valid_statuses.include?(target)
-            GenericStatus.new(target)
+            if 5 < ::Rails::VERSION::MAJOR ||
+               (::Rails::VERSION::MAJOR == 5 && 2 <= ::Rails::VERSION::MINOR)
```

Comment:
This is not quite what I meant, I meant conditionally defining the method on the status itself to match the new names, so that we're not doing version checks all the time.

Reply:
I don't like version checks. But I also don't like breaking tests for a whole lot of people because of a name change. I think that would mean at least a minor version upgrade, not a maintenance patch. I'm trying to get by with a maintenance patch. But perhaps I'm not understanding well the strategy you would prefer, and I'm entirely missing the boat.